src/ripple/rpc/handlers/AMMInfo.cpp

```diff
@@ -115,10 +137,19 @@ doAMMInfo(RPC::JsonContext& context)
         }
     }
 
-    auto const ammKeylet = keylet::amm(issue1, issue2);
+    auto const ammKeylet = [&]() {
+        if (isAssets)
+            return keylet::amm(issue1, issue2);
+        return keylet::amm(ammID);
+    }();
     auto const amm = ledger->read(ammKeylet);
     if (!amm)
         return rpcError(rpcACT_NOT_FOUND);
+    if (isAMMAccount)
+    {
+        issue1 = (*amm)[sfAsset];
+        issue2 = (*amm)[sfAsset2];
+    }
```

Comment:
This whole section is sued to get the `accountID`, `issue1` `issue2` and `amm`. Consider re-writing this with a lambda. Something like this (this incorporates the comments above as well):
```
Json::Value
doAMMInfo(RPC::JsonContext& context)
{
    auto const& params(context.params);
    Json::Value result;

    std::shared_ptr<ReadView const> ledger;
    result = RPC::lookupLedger(ledger, context);
    if (!ledger)
        return result;

    struct ValuesFromContextParams
    {
        std::optional<AccountID> accountID;
        Issue issue1;
        Issue issue2;
        std::shared_ptr<SLE const> amm;
    };

    auto getValuesFromContextParams =
        [&]() -> Expected<ValuesFromContextParams, error_code_i> {
        std::optional<AccountID> accountID;
        std::optional<Issue> issue1;
        std::optional<Issue> issue2;
        std::optional<uint256> ammID;

        if (params.isMember(jss::asset) && params.isMember(jss::asset2))
        {
            if (auto const i = getIssue(params[jss::asset], context.j))
                issue1 = *i;
            else
                return Unexpected(i.error());

            if (auto const i = getIssue(params[jss::asset2], context.j))
                issue2 = *i;
            else
                return Unexpected(i.error());
        }

        if (params.isMember(jss::amm_account))
        {
            auto const id = getAccount(params[jss::amm_account], result);
            if (!id)
                return Unexpected(rpcACT_MALFORMED);
            auto const sle = ledger->read(keylet::account(*id));
            if (!sle)
                return Unexpected(rpcACT_MALFORMED);
            ammID = sle->getFieldH256(sfAMMID);
        }

        if (issue1.has_value() == ammID.has_value())
            return Unexpected(rpcINVALID_PARAMS);

        if (params.isMember(jss::account))
        {
            accountID = getAccount(params[jss::account], result);
            if (!accountID || !ledger->read(keylet::account(*accountID)))
                return Unexpected(rpcACT_MALFORMED);
        }

        auto const ammKeylet = [&]() {
            if (issue1 && issue2)
                return keylet::amm(*issue1, *issue2);
            assert(ammID);
            return keylet::amm(*ammID);
        }();
        auto const amm = ledger->read(ammKeylet);
        if (!amm)
            return Unexpected(rpcACT_NOT_FOUND);
        if (!issue1 && !issue2)
        {
            issue1 = (*amm)[sfAsset];
            issue2 = (*amm)[sfAsset2];
        }

        return ValuesFromContextParams{
            accountID, *issue1, *issue2, std::move(amm)};
    };

    auto r = getValuesFromContextParams();
    if (!r)
    {
        RPC::inject_error(r.error(), result);
        return result;
    }

    auto& [accountID, issue1, issue2, amm] = *r;

```